hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/schema/TableResolverImpl.java

```diff
@@ -59,11 +61,21 @@ public class TableResolverImpl implements TableResolver {
             asList(CATALOG, SCHEMA_NAME_PUBLIC)
     );
 
+    private static final List<BiFunction<List<Mapping>, List<View>, Table>> ADDITIONAL_TABLE_PRODUCERS = Arrays.asList(
+            (m, v) -> new TablesTable(CATALOG, SCHEMA_NAME_INFORMATION_SCHEMA, SCHEMA_NAME_PUBLIC, m, v),
+            (m, v) -> new MappingsTable(CATALOG, SCHEMA_NAME_INFORMATION_SCHEMA, SCHEMA_NAME_PUBLIC, m),
+            (m, v) -> new MappingColumnsTable(CATALOG, SCHEMA_NAME_INFORMATION_SCHEMA, SCHEMA_NAME_PUBLIC, m, v),
+            (m, v) -> new ViewsTable(CATALOG, SCHEMA_NAME_INFORMATION_SCHEMA, SCHEMA_NAME_PUBLIC, v)
+    );
+
     private final NodeEngine nodeEngine;
     private final TablesStorage tableStorage;
     private final SqlConnectorCache connectorCache;
     private final List<TableListener> listeners;
 
+    private volatile int lastMappingsSize;
+    private int lastViewsSize;
```

Comment:
I believe it doesn't.
Write part:
```java
        this.lastViewsSize = views.size();
        this.lastMappingsSize = mappings.size(); // release
```
Read part:
```java
        int lastMappingsSize = this.lastMappingsSize; // acquire
        int lastViewsSize = this.lastViewsSize;
```
I agree that you can see those values from two different executions of the ```getTables```, but it doesn't break anything, it is just an initial size of the List, it can grow anyway if there is a new mapping/view added between two ```getTables```.